packages/web/lib/browser.js

```diff
@@ -200,49 +197,76 @@ const getBrowserScope = (parent) => ({
               query: match.query,
             });
 
+            // TODO: add checks to the ruleSet to identify if the url has
+            // changed. If so, and there are more rules to execute, throw error.
+            // If it's the last rule, then call `visit` on the new url so
+            // pageDefs are resolved once more
             await evalRuleSet(ruleSet, newPageScope);
 
-            // TODO: check if the url has changed? If so, recurse and execute the necessary `page` rule
-            const data = newPageScope.internal.data;
-            if (Object.keys(data).length) {
-              if (data.url) {
-                console.warn(
-                  "WARNING:: The key 'url' will always be overridden by the default url value (See docs <https://....>)"
+            if (match.scrape) {
+              const data = newPageScope.vars;
+              if (Object.keys(data).length) {
+                if (data.url !== href) {
```

Comment:
Since we used _links for 'links' maybe we should use _url for 'url' and change the error message to be the same as the one for _links to stay consistent.

Reply:
I was thinking `url` should specifically be special cased, and not in `_` since it's going to be used as the primary key for database outputs

Also, instead of _links, we could do `_browse` as an object that has links and any other meta inside that. Then the outputs have to do something similar to what stitch does when it tries to coerce JSON data into a SQL db

It isn't the most elegant thing though, but even stitch uses `__sdc_` keys 🤷